zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnCompensationSubscriptionBehaviour.java

```diff
@@ -80,6 +83,26 @@ public void createCompensationSubscription(
     }
   }
 
+  private boolean isMultiInstanceActivityAlreadySubscribed(
+      final ExecutableActivity element, final BpmnElementContext context, final String elementId) {
+    final var availableTasks =
+        List.of(
+            BpmnElementType.TASK,
+            BpmnElementType.SERVICE_TASK,
+            BpmnElementType.MANUAL_TASK,
+            BpmnElementType.USER_TASK,
+            BpmnElementType.SEND_TASK,
+            BpmnElementType.SCRIPT_TASK);
+
+    return BpmnElementType.MULTI_INSTANCE_BODY.equals(element.getFlowScope().getElementType())
+        && availableTasks.contains(element.getElementType())
+        && compensationSubscriptionState
+            .findSubscriptionsByProcessInstanceKey(
+                context.getTenantId(), context.getProcessInstanceKey())
+            .stream()
+            .anyMatch(sub -> elementId.equals(sub.getRecord().getCompensableActivityId()));
+  }
+
```

Comment:
:x: This is not the way. The check works for a multi-instance activity. However, it fixes the symptom and not the root. 

The underlying issue is that the multi-instance and the inner activity have the same boundary events. This is not correct. Only the multi-instance body should have the boundary events. 

We should fix the issue in [MultiInstanceActivityTransformer](https://github.com/camunda/zeebe/blob/main/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/MultiInstanceActivityTransformer.java#L56-L64) and remove the boundary events from the inner activity.

Reply:
After fixing the transformer, we need to adjust the logic for finding the boundary event ([ref](https://github.com/camunda/zeebe/blob/d54c0fe6060e44ce3c69e15e52d1291ab453d939/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnCompensationSubscriptionBehaviour.java#L233-L251)). 

The following code will return the inner activity but not the multi-instance body.  

```
processState.getFlowElement(
            context.getProcessDefinitionKey(),
            context.getTenantId(),
            BufferUtil.wrapString(elementId),
            ExecutableActivity.class);
```

To get the body, we could access the flow scope of the inner activity.